plugins/camera/camera.h

```diff
@@ -0,0 +1,298 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+#include <map>
+#include "plugin_base.h"
+
+namespace dronecore {
+
+class CameraImpl;
+class Device;
+
+/**
+ * @brief The Camera class can be used to manage cameras that implement the
+ * MAVLink Camera Protocol: https://mavlink.io/en/protocol/camera.html.
+ *
+ * The plugin needs to be instantiated multiple times for multiple cameras.
+ * However, this is not supported yet.
+ *
+ * Synchronous and asynchronous variants of the camera methods are supplied.
+ */
+class Camera : public PluginBase
+{
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific Device.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto camera = std::make_shared<Camera>(device);
+     *     ```
+     *
+     * @param device The specific device associated with this plugin.
+     */
+    explicit Camera(Device &device);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~Camera();
+
+    /**
+     * @brief Possible results returned for camera commands.
+     */
+    enum class Result {
+        SUCCESS = 0,
+        IN_PROGRESS,
+        BUSY,
+        DENIED,
+        ERROR,
+        TIMEOUT,
+        WRONG_ARGUMENT,
+        UNKNOWN
+    };
+
+    /**
+     * @brief Returns a human-readable English string for Camera::Result.
+     *
+     * @param result The enum value for which a human readable string is required.
+     * @return Human readable string for the Camera::Result.
+     */
+    static const char *result_str(Result result);
+
+    /**
+     * @brief Callback type for asynchronous Camera calls.
+     */
+    typedef std::function<void(Result)> result_callback_t;
+
+    /**
+     * @brief Take photo (synchronous).
+     *
+     * This takes one photo immediately.
+     *
+     * @return Result of request.
+     */
+    Result take_photo();
+
+    /**
+     * @brief Start photo interval (synchronous).
+     *
+     * Starts a photo timelapse given an interval.
+     *
+     * @param interval_s The interval between photos in seconds.
+     * @return Result of request.
+     */
+    Result start_photo_interval(float interval_s);
+
+    /**
+     * @brief Stop photo interval (synchronous).
+     *
+     * Stops a photo timelapse again.
+     *
+     * @return Result of request.
+     */
+    Result stop_photo_interval();
+
```

Comment:
How about providing a static method that tells the number of cameras on the drone.
```
static size_t Camera::how_many_cameras() const;
```
OR
```
```
static size_t Camera::total_cameras() const;
```
whatever is more intuitive.

Reply:
As said, I'll work on multi camera support when single camera support is merged and properly working. I can't do all at once :smile:.

Reply:
Lazy :-)

Reply:
I have a gazillion things I could do every day and I'm always trying to figure out what's most relevant :confused:.

Reply:
Just teasing (I thought obviously?). You do plenty of good.